src/cascadia/TerminalSettingsModel/TerminalSettings.cpp

```diff
@@ -183,13 +183,49 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         return settingsPair;
     }
 
-    void TerminalSettings::_ApplyAppearanceSettings(const IAppearanceConfig& appearance, const Windows::Foundation::Collections::IMapView<winrt::hstring, ColorScheme>& schemes)
+    void TerminalSettings::_ApplyAppearanceSettings(const IAppearanceConfig& appearance,
+                                                    const Windows::Foundation::Collections::IMapView<winrt::hstring, ColorScheme>& schemes,
+                                                    const winrt::Microsoft::Terminal::Settings::Model::Theme currentTheme)
     {
         _CursorShape = appearance.CursorShape();
         _CursorHeight = appearance.CursorHeight();
-        if (!appearance.ColorSchemeName().empty())
+
+        const auto requestedTheme = currentTheme.RequestedTheme();
+        const auto defaultName = appearance.ColorSchemeName();

```

Comment:
Yeah, I don't think we need `defaultName` (`ColorSchemeName()`) anymore. I see that you're keeping it around so that the UI has a value it's in charge of, but I think the right move there should be that the TerminalSettingsEditor sets `DarkColorSchemeName` and `LightColorSchemeName` when it's set via the UI.